pulser-core/pulser/sequence/sequence.py

```diff
@@ -205,21 +213,45 @@ def declared_variables(self) -> dict[str, Variable]:
     @property
     def available_channels(self) -> dict[str, Channel]:
         """Channels still available for declaration."""
-        # Show all channels if none are declared, otherwise filter depending
-        # on whether the sequence is working on XY mode
-        # If already in XY mode, filter right away
-        if not self._schedule and not self._in_xy:
-            return dict(self._device.channels)
+        all_channels = {**self._device.channels, **self._device.dmm_channels}
+        xy_channels = {
+            id: ch for id, ch in all_channels.items() if ch.basis == "XY"
+        }
+        if not self._schedule:
+            # Show all channels if none are declared
+            if not self._in_xy:
+                return all_channels
+            # If already in XY mode, filter right away but let the possibility
+            # to define an SLM Mask using a DMM channel
+            return {**xy_channels, **self._device.dmm_channels}
```

Comment:
I agree that the DMM channels should appear as available, but I don't think it should be possible to declare them in XY mode